src/common/src/config.rs

```diff
@@ -1007,6 +1007,8 @@ pub struct S3ObjectStoreConfig {
     pub retry_unknown_service_error: bool,
     #[serde(default)]
     pub developer: S3ObjectStoreDeveloperConfig,
+    #[serde(default = "default::object_store_config::s3::identity_resolution_timeout_s")]
+    pub identity_resolution_timeout_s: u64,
```

Comment:
minor style issue: let's keep `developer` as the last field and move this before `developer`